macaroonbakery/tests/test_agent.py

```diff
@@ -90,12 +101,289 @@ def test_load_agents_into_cookies(self):
+            try:
+                auth_checker.allow(ctx, test_ops)
+                resp = response(status_code=200,
+                                content='done')
+            except macaroonbakery.PermissionDenied:
+                caveats = [
+                    checkers.Caveat(location='http://0.3.2.1',
+                                    condition='is-ok')
+                ]
+                m = server_bakery.oven.macaroon(
+                    version=macaroonbakery.LATEST_BAKERY_VERSION,
+                    expiry=datetime.utcnow() + timedelta(days=1),
+                    caveats=caveats, ops=test_ops)
+                content, headers = httpbakery.discharge_required_response(
+                    m, '/',
+                    'test',
+                    'message')
+                resp = response(status_code=401,
+                                content=content,
+                                headers=headers)
+            return request.hooks['response'][0](resp)
+
+        @urlmatch(path='.*/discharge')
+        def discharge(url, request):
+            qs = parse_qs(request.body)
+            if qs.get('token64') is None:
+                return response(
+                    status_code=401,
+                    content={
+                        'Code': httpbakery.ERR_INTERACTION_REQUIRED,
+                        'Message': 'interaction required',
+                        'Info': {
+                            'InteractionMethods': {
+                                'agent': {'login-url': '/login'},
+                            },
+                        },
+                    },
+                    headers={'Content-Type': 'application/json'})
+            else:
+                qs = parse_qs(request.body)
+                content = {q: qs[q][0] for q in qs}
+                m = httpbakery.discharge(checkers.AuthContext(), content,
+                                         discharge_key, None, None)
+                return {
+                    'status_code': 200,
+                    'content': {
+                        'Macaroon': m.serialize_json()
+                    }
+                }
+
+        key = macaroonbakery.generate_key()
+
+        @urlmatch(path='.*/login')
+        def login(url, request):
+            b = macaroonbakery.Bakery(key=discharge_key)
+            m = b.oven.macaroon(
+                version=macaroonbakery.LATEST_BAKERY_VERSION,
+                expiry=datetime.utcnow() + timedelta(days=1),
+                caveats=[macaroonbakery.local_third_party_caveat(
+                    key.public_key,
+                    version=httpbakery.request_version(request.headers))],
+                ops=[macaroonbakery.Op(entity='agent', action='login')])
+            return {
+                'status_code': 200,
+                'content': {
+                    'macaroon': m.to_dict()
+                }
+            }
+
+        with HTTMock(server_get):
```

Comment:
You can do multiple `with`s in the same statement I think:

    with HTTMock(server_get), HTTMock(discharge), HTTMock(login):

Reply:
done